Source/Core/VideoCommon/PixelShaderGen.cpp

```diff
@@ -1675,8 +1675,13 @@ static void WriteAlphaTest(ShaderCode& out, const pixel_shader_uid_data* uid_dat
   else
     out.Write(")) {{\n");
 
-  out.Write("\t\tocol0 = float4(0.0, 0.0, 0.0, 0.0);\n");
-  if (use_dual_source && !(api_type == APIType::D3D && uid_data->uint_output))
+  if (uid_data->uint_output)
+    out.Write("\t\tocol0 = uint4(0, 0, 0, 0);\n");
+  else
+    out.Write("\t\tocol0 = float4(0.0, 0.0, 0.0, 0.0);\n");
+  if (use_dual_source && uid_data->uint_output)
+    out.Write("\t\tocol1 = uint4(0, 0, 0, 0);\n");
+  else if (use_dual_source)
     out.Write("\t\tocol1 = float4(0.0, 0.0, 0.0, 0.0);\n");
```

Comment:
I'd prefer this since it makes it more obvious that both cases are for `use_dual_source`:

```C++
if (use_dual_source)
{
  if (uid_data->uint_output)
    out.Write("\t\tocol1 = uint4(0, 0, 0, 0);\n");
  else
    out.Write("\t\tocol1 = float4(0.0, 0.0, 0.0, 0.0);\n");
}
```